packages/p2p/src/components/app.jsx

```diff
@@ -65,7 +87,35 @@ const App = () => {
             }
         });
 
-        return () => general_store.onUnmount();
+        if (/\/p2p$/.test(location.pathname)) {
+            history.push(routes.p2p_buy_sell);
+            general_store.setActiveIndex(0);
+        } else if (/\/orders$/.test(location.pathname)) {
+            history.push(routes.p2p_orders);
+            general_store.setActiveIndex(1);
+        } else if (/\/my-ads$/.test(location.pathname)) {
+            history.push(routes.p2p_my_ads);
+            general_store.setActiveIndex(2);
+        } else if (/\/my-profile$/.test(location.pathname)) {
+            history.push(routes.p2p_my_profile);
+            general_store.setActiveIndex(3);
+        } else if (/\/advertiser$/.test(location.pathname)) {
+            if (location.search || general_store.counterparty_advertiser_id) {
+                const url_params = new URLSearchParams(location.search);
```

Comment:
**suggestion (non-blocking)**: We might need to sanitise this param later, or else the code scanning for security will mark this as client-side URL redirect warnings later

Reply:
I tested this, codeQL passed in my fork, I guess we can ignore this